Create a unit test for Update Device Service LastReported by ID.

akramtexas opened the pull request:
Fix #1737

AnthonyMBonafide commented on internal/core/metadata/rest_deviceservice_test.go:
```diff
@@ -594,3 +691,20 @@ func createDeviceServiceRequestWithBody(
+	dbMock.On("UpdateDeviceService", mock.Anything).Return(testError)
+
+	return dbMock
+}
```
Can this be removed in favor of refactoring `createMockWithOutlines` to support mocking methods with varying number or return entities? It seems like we have outgrown the original function and it needs to be updated to support this. If we do want to keep these functions I think we should rename them better so it is clear what the difference is between them.

If we do refactor the `createMockWithOutlines` I think we can than 
 remove `TestUpdateServiceLastReportedByIdServiceUnavailable` in favor of a test table.

akramtexas replied:
I like the suggestion and will pursue the implementation 👍

AnthonyMBonafide commented on internal/core/metadata/rest_deviceservice_test.go:
```diff
@@ -560,6 +576,87 @@ func TestUpdateAdminStateByName(t *testing.T) {
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rr := httptest.NewRecorder()
+			var loggerMock = logger.NewMockClient()
```
For consistency, can we use the `:=` operator here. Our code base tends to favor `:=` over `var`. This also applies to other uses of `var` in these changes.

akramtexas commented:
@AnthonyMBonafide 
- Following up on your suggestion, I have refactored the func `createMockWithOutlines()` to support mocking methods with varying number of arguments and/or varying number of return entities. This resulted in updating about 50+ locations wherefrom `createMockWithOutlines()` is called. 
- Removed TestUpdateServiceLastReportedByIdServiceUnavailable() in favor of a table-driven test.
